test/truffle/claimManager.upgrade.test.ts

```diff
@@ -0,0 +1,19 @@
+const { deployProxy, upgradeProxy } = require('@openzeppelin/truffle-upgrades');
+const { assert } = require('chai');
+
+const ClaimManager = artifacts.require('ClaimManager');
+const ClaimManagerUpgradeTest = artifacts.require('ClaimManagerUpgradeTest');
+
+contract('ClaimManager Upgrade Test', async () => {
+    it('should upgrade contract', async () => {
+        const claimManager = await deployProxy(ClaimManager, ["0x84d0c7284A869213CB047595d34d6044d9a7E14A", "0xd7CeF70Ba7efc2035256d828d5287e2D285CD1ac"]);
+        const firstVersion = await claimManager.version();
+
+        assert.equal(firstVersion, 'v0.1');
+
+        const upgradedClaimManager = await upgradeProxy(claimManager.address, ClaimManagerUpgradeTest);
+        const secondVersion = await upgradedClaimManager.version();
+
+        assert.equal(secondVersion, 'v0.2');
```

Comment:
Do you think it would be worth adding additional tests around the data still being available post upgrade?

Reply:
This would require to deploy all the contracts (ERC1056, ENSRegistry and more). I initially tried to deploy those contracts but I was not able to, so for the case ERC1056 is not part of the contracts in IAM and thus the `truffle atifacts` doesn't fetch it and I tried using other libraries to create contract but it didn't happen. So, I settled for only basic upgrade test.

Reply:
Ok, I see. How is it working with Volta addresses? Is it actually deploying to Volta?

Reply:
Nope, so the addresses are plain string to ClaimManager contract and it doesn't verify at all. It simply passes those two string addresses to `initialize` method and deploy the contract on local network.

Reply:
Ah, but if you tried to call any actually behaviour it will not work, right?

Reply:
Correct!

Reply:
Ok, I think it would be better to use random addresses instead of the actual Volta addresses as one could assume that the addresses are meaningful if they match the Volta addresses (as I did 😆 ).